src/main/java/seedu/address/model/library/Library.java

```diff
@@ -62,7 +62,12 @@ public Library(int i) {
      * @param bookList The list of books to initialize the library with.
      */
     public Library(ObservableList<Book> bookList) {
-        this.bookList = bookList;
+        ObservableList<Book> expectedBookList = FXCollections.observableArrayList();
+        for (Book book : bookList) {
+            expectedBookList.add(book);
+        }
+        this.bookList = expectedBookList;
+        threshold = new Threshold();
```

Comment:
When fixing the test cases, we realised that the constructor does not return a new object. As a result, when making changes to the expected model, the changes also reflect in the original model. Thus, we made this change in order to have our test cases work as expected.